client/packages/common/src/hooks/useNativeClient/useNativeClient.ts

```diff
@@ -162,16 +160,10 @@ export const useNativeClient = ({
     if (!autoconnect) return;
     if (previousServer === null) return;
 
-    fetch(frontEndHostUrl(previousServer))
-      .then(response => {
-        if (response.status === 200) {
-          connectToServer(previousServer);
-        }
-      })
-      .catch(error => {
-        setState(state => ({ ...state, connectToPreviousTimedOut: true })),
-          console.error('Connecting to previous server:', error);
-      });
+    // this will check to see if the server is alive and if so, connect to it
+    connectToServer(previousServer)
```

Comment:
oh? I missed one - I thought I'd caught all of the usages. thanks!

Reply:
yes, though in this case we are only saving a single possible value of the previous server to preferences and autoconnect is static for the page. So while theoretically possible, I can't see how we could result in two promises running!

Have looked at react-query and I don't see how I can make use of `useQuery` as the parameters would need to be set at the point of instantiation rather than at the point of calling. For that we need `useMutation` which I think does the trick. [The documentation](https://tanstack.com/query/latest/docs/react/guides/mutations#consecutive-mutations) has that `onSuccess` and `onError` are called only once, and for the last call, so I've used those.

the usage in `DiscoveredServers` is more likely to have the race condition, I think. Have also implemented the same approach there. 🙏

Reply:
think `useMutate` would suffer from the same problem, i.e. you can have multiple request in parallel... Sorry was probably over-thinking since it can't really happen in practice.

You probably need something like redux to dispatch the response and then only update the state if the request is matching the current server/config state...